src/index.ts

```diff
@@ -58,18 +64,26 @@ export async function listMessages(
   before?: Date | undefined,
   after?: Date | undefined
 ): Promise<DecodedMessage[]> {
-  return (
-    await XMTPModule.loadMessages(
-      clientAddress,
-      [conversationTopic],
-      [conversationID],
-      limit,
-      before?.getTime,
-      after?.getTime
-    )
-  ).map((json: string) => {
-    return JSON.parse(json);
+  const messages = await XMTPModule.loadMessages(
+    clientAddress,
+    [conversationTopic],
+    [conversationID],
+    limit,
+    before?.getTime,
+    after?.getTime
+  );
+
+  const mappedMessages = messages.map((message) => {
+    return decode(message);
   });
+
+  for (const message of mappedMessages) {
+    const encodedContent = proto.content.EncodedContent.decode(message.content);
+
+    message.content = encodedContent;
+  }
+
+  return mappedMessages;
```

Comment:
Maybe we should move this into the `map` above so we don't need to loop over the message list twice?